internal/services/azurestackhci/stack_hci_virtual_hard_disk_resource_test.go

```diff
@@ -0,0 +1,265 @@
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clusterClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r StackHCIVirtualHardDiskResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_stack_hci_virtual_hard_disk" "test" {
+  name                = "acctest-vhd-%s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  custom_location_id  = %q
+  disk_size_in_gb     = 2
+
+  lifecycle {
+    ignore_changes = [storage_path_id]
+  }
+}
+`, template, data.RandomString, os.Getenv(customLocationIdEnv))
+}
+
+func (r StackHCIVirtualHardDiskResource) update(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_stack_hci_virtual_hard_disk" "test" {
+  name                = "acctest-vhd-%s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  custom_location_id  = %q
+  disk_size_in_gb     = 2
+
+  tags = {
+    foo = "bar"
+  }
+
+  lifecycle {
+    ignore_changes = [storage_path_id]
+  }
+}
+`, template, data.RandomString, os.Getenv(customLocationIdEnv))
+}
+
+func (r StackHCIVirtualHardDiskResource) updateTag(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_stack_hci_virtual_hard_disk" "test" {
+  name                = "acctest-vhd-%s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  custom_location_id  = %q
+  disk_size_in_gb     = 2
+
+  tags = {
+    env = "test"
+    foo = "bar"
+  }
+
+  lifecycle {
+    ignore_changes = [storage_path_id]
+  }
+}
+`, template, data.RandomString, os.Getenv(customLocationIdEnv))
+}
+
+func (r StackHCIVirtualHardDiskResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_stack_hci_storage_path" "test" {
+  name                = "acctest-sp-%[2]s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  custom_location_id  = %[3]q
+  path                = "C:\\ClusterStorage\\UserStorage_2\\sp-%[2]s"
+}
+
+resource "azurerm_stack_hci_virtual_hard_disk" "test" {
+  name                = "acctest-vhd-%[2]s"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  custom_location_id  = %[3]q
+  disk_size_in_gb     = 2
+  storage_path_id     = azurerm_stack_hci_storage_path.test.id
+
+  tags = {
```

Comment:
This is not quite "complete" as some of the attributes are not present yet.

Reply:
updated